src/shogun/machine/Machine.h

```diff
@@ -334,8 +334,40 @@ class CMachine : public CStoppableSGObject
 		 */
 		virtual bool train_machine(CFeatures* data=NULL)
 		{
-			SG_ERROR("train_machine is not yet implemented for %s!\n",
-					get_name());
+			bool result = false;
+			if (support_features_dispatching())
+			{
+				switch (data->get_feature_class())
+				{
+				case C_DENSE:
+					result = train_dense(data);
+					break;
+				case C_STRING:
+					result = train_string(data);
+					break;
+				default:
+					SG_ERROR(
+					    "train_machine is not yet implemented for %s!\n",
+					    get_name());
+				}
+			}
+			return result;
+		}
+
+		virtual bool train_dense(CFeatures* data)
+		{
+			SG_NOTIMPLEMENTED
+			return false;
+		}
+
+		virtual bool train_string(CFeatures* data)
+		{
+			SG_NOTIMPLEMENTED
+			return false;
+		}
+
+		virtual bool support_features_dispatching()
```

Comment:
I am afraid we need one of such checks for each feature class to avoid problems when passing for example passing string features to LARS